### ISSUE TITLE
How to use Convolution1D layer to connect the FullConnection layer,Flatten() doesn't work

### ISSUE THREAD

**Issue by saicoco** [fchollet/keras#2483]
Please make sure that the boxes below are checked before you submit your issue. Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by saicoco** [comment]
I just wanna use the Convolution1D layer to process audio signal to classify music,there are some Conv1D layes before the Fullconnection layer , when I use the Flatten(), 
`
model=Sequential()
    model.add(Convolution1D(nb_filter=32, filter_length=8, border_mode='valid',input_dim=128))
    model.add(Activation('sigmoid'))
    model.add(MaxPooling1D(pool_length=4))
    model.add(Convolution1D(nb_filter=32, filter_length=8, border_mode='valid'))
    model.add(Activation('sigmoid'))
    model.add(MaxPooling1D(pool_length=4))
    model.add(Flatten())
    model.add(Dense(100))
    model.add((Activation('sigmoid')))
    model.add(Dense(50))
    model.add((Activation('sigmoid')))
    model.add(Dense(6))
    model.add(Activation('softmax'))
    sgd = SGD(lr=0.1, momentum=0.9, nesterov=True, decay=1e-6)
    model.compile(optimizer=sgd, loss='categorical_crossentropy', metrics=['accuracy'])
`
just like this model, what should I do to connect the DenseLayer and maxpool layer? Thanks very much

**Comment by joelthchao** [comment]
In the document
> input_length: Length of input sequences, when it is constant. This argument is required if you are going to connect Flatten then Dense layers upstream (without it, the shape of the dense outputs cannot be computed).

You should specify `input_length` argument like:

`model.add(Convolution1D(nb_filter=32, filter_length=8, border_mode='valid', input_dim=128, input_length=your_input_length))`

**Comment by saicoco** [comment]
Thanks @joelthchao, 
Raise `Exception: The shape of the input to "Flatten" is not fully defined (got (0, 32). Make sure to pass a complete "input_shape" or "batch_input_shape" argument to the first layer in your model.`
Because of the output_shape of `Convolutio1D` is `3D tensor with shape: (samples, new_steps, nb_filter)`
my data_shape=(81805, 128), labels_shape=(81805, 128),I just doubt whether my data is ok.I'm noob in audio process

**Comment by joelthchao** [comment]
Do you mean you have 81805 data points, and each data point is a 128-dim vector?
And your task is going to classify each data point into 6 class?

**Comment by saicoco** [comment]
@joelthchao ,yes. when I reshape the data to new shape,like` (new_sample, length, dim),  
old_samples = new_sample * length`  
 `model.add(Convolution1D(nb_filter=32, filter_length=8, border_mode='valid', input_dim=(length, dim))`,the model works well. Thanks all the same,I just do not understand the document

**Comment by zhangxulong** [comment]
> yes.when I reshape the data to new shape,like(new_sample, length, dim), old_samples = new_sample * length
model.add(Convolution1D(nb_filter=32, filter_length=8, border_mode='valid', input_dim=(length, dim)),the model works well. Thanks all the same,I just do not understand the document completely

**if you change new_sample size it should change you Y data label ** so I want know how is your program
